datalad/support/gitrepo.py

```diff
@@ -937,14 +937,18 @@ def clone(cls, url, path, *args, **kwargs):
         # make sure that Git doesn't mangle relative path specification into
         # mildly obscure absolute paths
         # https://github.com/datalad/datalad/issues/3538
+        # Note, that POSIX is required even on windows, since it's an URL!
         if isinstance(url_ri, PathRI):
             path = Path(url)
             if not path.is_absolute():
-                # always in POSIX even on windows
-                path = path.as_posix()
-                if not path.startswith(op.pardir + '/'):
-                    path = posixpath.join(op.curdir, path)
-                gr.config.set('remote.origin.url', path,
+                # get git-created path
+                git_url = gr.config.get('remote.origin.url')
+                # Note: Not sure, whether there are circumstances where this is relative already
+                if op.isabs(git_url):
+                    # ... and make it a relative one
+                    # Note: Using posixpath here, since pathlib's relative_to isn't what you'd expect
```

Comment:
As I mentioned on the call, I'm a bit worried that pathlib is avoiding relpath's more extensive behavior for some edge case where it can behave incorrectly.  But I can't come up with what that would be, and looking at pathlib's docs and source doesn't reveal any hints, so that worry is probably unfounded.

Reply:
Well, there has to be a reason for pathlib to provide such limited functionality with `relative_to`. But I can't come up with a scenario where `os.path.relpath` wouldn't work here either. May be there's a way to break it, when the path to the repository (our starting point for the relative path) would contain symlinks. But that doesn't apply here, since `GitRepo.path` is realpath'd already.